File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> freq(26);
        if(s.size()!=t.size()) return false;
        for(int i=0;i<s.size();i++){
            freq[s[i]-'a']++;
            freq[t[i]-'a']--;
        }
        for(int i=0;i<26;i++){
            if(freq[i]!=0) return false;
        }
        return true;
    }
};

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length()!=t.length()) return false;
        Map<Character,Integer> freq = new HashMap<>();
        for(int i=0;i<s.length();i++){
            char ch1 = s.charAt(i), ch2 = t.charAt(i);
            freq.put(ch1,freq.getOrDefault(ch1,0)+1);
            freq.put(ch2,freq.getOrDefault(ch2,0)-1);
        }
        for(int value: freq.values()){
            if(value!=0){
                return false;
            }
        }
        return true;
    }
}